src/modules/navigator/mission_block.cpp

```diff
@@ -282,6 +282,47 @@ MissionBlock::is_mission_item_reached()
 				}
 			}
 
+		} else if (_mission_item.nav_cmd == NAV_CMD_CONDITION_GATE) {
+
+			// default to true as a number of geometric corner cases
+			// might prevent reaching the gate - we only set this to false
+			// if we are able to show that the gate conditions are feasible
+			bool gate_reached = true;
+
+			struct position_setpoint_s *curr_sp = &_navigator->get_position_setpoint_triplet()->current;
+
+			// if either is not valid the gate is reached
+			if (curr_sp->valid) {
+
+				// zero is the current gate position
+				matrix::Vector2f gate_sp(0, 0);
+				struct map_projection_reference_s ref_pos;
+				map_projection_init(&ref_pos, _mission_item.lat, _mission_item.lon);
+
+				matrix::Vector2f gate_to_curr_sp;
+				map_projection_project(&ref_pos, curr_sp->lat, curr_sp->lon, &gate_to_curr_sp(0), &gate_to_curr_sp(1));
+				matrix::Vector2f vehicle_pos;
+				map_projection_project(&ref_pos, _navigator->get_global_position()->lat, _navigator->get_global_position()->lon,
+						       &vehicle_pos(0), &vehicle_pos(1));
+				float res = vehicle_pos.dot(gate_to_curr_sp.normalized());
```

Comment:
I would call the float `dot_product` instead of the generic `res`. And you can always make it `const` if it's just calculated once.

Reply:
Still applies...

Reply:
Fair.